### PR TITLE
Created filter component

### LINKED ISSUE #46: [React] filter component in cars page
In this task you should create filter component with states that we need.

### PR DESCRIPTION
**Pull request by tariqabusada**
Create filter component that will be filter the cars by multiple filter type

### CODE REVIEW

File: client/src/components/CarsFilter/index.tsx [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+        'Adaptive cruis control',
+      ],
+      description: '',
+      fuel: 'petrol',
+      customerId: 2,
+    },
+  ];
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const brands = ['Toyota',
```
Comment (LinaYahya) [inline client/src/components/CarsFilter/index.tsx]: We already have a `brand.json` file, So try to use data from there, also create a file for `models` and export data from there

File: client/src/components/CarsFilter/index.tsx [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+import {
+  Autocomplete, TextField, Switch, Slider,
+} from '@mui/material';
+import {
+  useState, SetStateAction, useEffect,
+} from 'react';
+import './style.css';
+
+function CarsFilter() {
+  const mileRang = [
```
Comment (LinaYahya) [inline client/src/components/CarsFilter/index.tsx]: no need to have these data at the component, as on every render these object will create from the strat,
SO move it to separate file and use it here,
The same for all data you used and `getYears` function

File: client/src/components/CarsFilter/index.tsx [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+    'IS350 3.5L V6', 'HS 2.0L TC I4',
+    'Model Y Standard',
+    'Mustang 2.3L i4 EcoBoost',
+    'A8 6.3L W12',
+    'Wrangler 3.6L V6', 'Huracan 5.2 V10',
+    'QX70 3.7L V6'];
+    /* const getArray = (array: Array<any>) => {
+      const newArray = array.map((c) => c.model).sort(
+        (a, b) => b - a,
+      );
```
Comment (LinaYahya) [inline client/src/components/CarsFilter/index.tsx]: remove all comments you have

File: client/src/components/CarsFilter/index.tsx [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+          )}
+        />
+        <TextField
+          label="Max price"
+          sx={{ width: 250 }}
+          type="number"
+          onChange={changeMaxPrice}
+          value={!maxPrice ? '' : maxPrice}
+        />
+        {' '}
```
Comment (LinaYahya) [inline client/src/components/CarsFilter/index.tsx]: add a label for switch

File: client/src/components/CarsFilter/index.tsx [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+  useEffect(() => {
+    // here we will fetch the data and set it on a props that it defined in parent component
+    setData(data);
+    console.log(data);
+  }, [model, mileage, year, fuel, isGoodPrice]);
+
+  console.log(model, mileage, isGoodPrice, fuel, maxPrice);
+
+  return (
+    <section className="filter">
```
Comment (LinaYahya) [inline client/src/components/CarsFilter/index.tsx:87]: try to style the side filter as inputs didn't have vertical margin

File: client/src/components/CarsFilter/index.tsx [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+    }
+    return carsArray.map((c) => c[filter]).sort(
+      (a, b) => b - a,
+    );
+    } */
+  const changePriceType = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setPriceBool(event.target.checked);
+  };
+
+  const changeModel = (e: any, values: SetStateAction<any>) => {
```
Comment (LinaYahya) [inline client/src/components/CarsFilter/index.tsx:65]: change all your `any` types

**Review by LinaYahya — CHANGES_REQUESTED** [review]
Good Job @tariqabusada

**Review by husamkamal — APPROVED** [review]

File: client/src/components/CarsFilter/index.tsx [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+
+  return (
+    <section className="filter">
+      <fieldset className="filters">
+        <p className="title">Cars filter</p>
+        <Autocomplete
+          disablePortal
+          id="combo-box"
+          onChange={changeBrand}
+          options={brands.map((e) => e.brand)}
```
Comment (LinaYahya) [inline client/src/components/CarsFilter/index.tsx:94]: give it a default value as state.brand
```suggestion
          options={brands.map((e) => e.brand)}
           defaultValue={state?.brand || null}

```

File: client/src/components/CarsFilter/index.tsx [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+  for (let i = 1990; i <= yearNow; i += 1) {
+    yearsArr.push(`${i}`);
+  }
+  return yearsArr;
+};
+
+function CarsFilter() {
+  const { state } = useLocation();
+
+  const [brand, setBrand] = useState(state.brand);
```
Comment (LinaYahya) [inline client/src/components/CarsFilter/index.tsx]:
```suggestion
  const [brand, setBrand] = useState('');
```

File: client/src/components/CarsFilter/index.tsx [hunk 9]
```diff
@@ -0,0 +1,277 @@
...
+    value: 100,
+    label: '1M',
+  },
+];
+
+const cars = [
+  {
+    brand: 'Toyota',
+    model: 'Land Cruiser VXR 4.0L V6',
+    price: 48000,
```
Comment (LinaYahya) [inline client/src/components/CarsFilter/index.tsx]: remove cars from here

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by LinaYahya — APPROVED** [review]

**Review by abdou059 — APPROVED** [review]

**Pull request merged**